### ISSUE TITLE
Allocation lost after nomad restart or system reboot

### ISSUE THREAD

**Issue by reiyanyan** [hashicorp/nomad#24557]
### Nomad version
Nomad v1.9.3
BuildDate 2024-11-11T16:35:41Z
Revision d92bf1014886c0ff9f882f4a2691d5ae8ad8131c

### Operating system and Environment details
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.5 LTS
Release:	22.04
Codename:	jammy

### Issue
Nomad allocation are gone after restarting client or nomad service

### Reproduction steps
- Deploy an simple jobs
- restarting nomad.service or reboot system
- all jobs are gone after nomad UI been healthy again

#### Expected Result
- All jobs are still exist or re-run after restarting nomad.service or reboot system

#### Actual Result
- All jobs are gone

### Job file (if appropriate)
```

job "job-one" {
  group "group-web" {
    network {
      mode = "bridge"
      port "http" {
	      static = 80
        to = 80
      }
    }

    service {
      provider = "nomad"
      name = "test-web"
      port = "http"
    }

    task "app" {
      driver = "docker"
      config {
        image      = "nginx:latest"
        ports      = ["http"]
      }
    }
  }
}
```

**Comment by jrasell** [comment]
Hi @reiyanyan and thanks for raising this issue. I have not been able to reproduce this, could you provide some more details of your Nomad configuration please?

Test machine:
```console
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 12 (bookworm)
Release:	12
Codename:	bookworm
```

Nomad version:
```console
$ nomad version
Nomad v1.9.4-dev
BuildDate 2024-11-25T09:18:07Z
Revision 629e869a752c152f2675731652a52bd6c83b46fe
```

Nomad config:
```hcl
data_dir   = "/tmp/nomad-all-in-one-dev/"

log_level            = "DEBUG"
log_include_location = true
enable_debug         = true
region               = "europe-west-1"

telemetry {
  in_memory_collection_interval = "1s"
  in_memory_retention_period    = "10s"
  publish_allocation_metrics    = true
  publish_node_metrics          = true
  prometheus_metrics            = true
}

client {
  enabled = true

  server_join {
    retry_join = ["127.0.0.1:4647"]
  }
}

server {
  authoritative_region = "europe-west-1"
  enabled              = true
  bootstrap_expect     = 1
}

acl {
  enabled = true
}
```

I ran the job you provided and then completely stopped the Nomad process,; then started it again a few moments later. The allocation detail is seen below from before and after the stop/start.

Before stop:
```console
Allocations
ID        Node ID   Task Group  Version  Desired  Status   Created  Modified
e7728363  f7ffe2c1  group-web   0        run      running  18s ago  1s ago
```

After stop/start:
```console
Allocations
ID        Node ID   Task Group  Version  Desired  Status   Created    Modified
e7728363  f7ffe2c1  group-web   0        run      running  6m34s ago  6m17s ago

```

**Comment by reiyanyan** [comment]
@jrasell i'am using default configuration that stored in /etc/nomad/nomad.hcl

**Comment by ocharles** [comment]
@reiyanyan You need to give more information than that. What is your OS? What are the steps to reproduce this? Can you reproduce this reliably or did this only happen once? Rather than saying you're using the default it would be better to paste in what you literally have, so there's no confusion or guessing.

**Comment by reiyanyan** [comment]
here's my nomad configuration that i use
```
data_dir  = "/opt/nomad/data"
bind_addr = "0.0.0.0"
datacenter = "dc1"

# Enable the server
server {
  enabled          = true
  bootstrap_expect = 1
}

# Enable the client
client {
  enabled = true
}
```

**Comment by jrasell** [comment]
Hi @reiyanyan, thanks for clarifying.

I ran another test where I installed Nomad using [these instructions](https://developer.hashicorp.com/nomad/docs/install) and started the service using the default configuration file placed which matches the one you're using. I again was unable to reproduce the behaviour you're describing. In my environment I see the expected behaviour where the client workloads continue to run when the Nomad process is stopped, and then Nomad reattaches to them when it starts back up.

Are you using a non-standard service file to control Nomad? When you restart Nomad, is it possible another process clearing out the data directory?
